lib/kitchen/driver/ec2.rb

```diff
@@ -134,15 +133,34 @@ def default_public_ip_association
 
       private
 
+      def aws_credentials
+        if config[:use_iam_profile]
+          {
+            :use_iam_profile => true
+          }
+        else
+          credentials = {}
+          [:aws_access_key_id, :aws_secret_access_key].each { |key| 
+            if !config.key?(key) or config[key].nil? or config[key].to_s.empty?
+              raise Kitchen::UserError, "Must provide #{key} or use IAM profile"
```

Comment:
yah, there's just a lot of conditions so the line would be really long (cross-referenced some validation being done in kitchen-ec2 elsewhere in their native validation features, which I'm not using because in this case, validation depends on other config keys)